### PR TITLE
[4.0] [com_languages] Parameterized queries

### PR DESCRIPTION
**Pull request by SharkyKZ**
### Summary of Changes

Adds parameterized queries and cleans up SQL in `com_languages`.

### Testing Instructions

Set up a multilingual site. Use `com_languages` in all sorts of way, e.g.:

Search, install languages.
Create/modify content languages.
Create language overrides.
Check multilanguage status.

### Expected result

Works like before.

### Documentation Changes Required

No.

### CODE REVIEW

File: administrator/components/com_languages/Helper/MultilangstatusHelper.php [hunk 1]
```diff
@@ -158,7 +165,7 @@ public static function getContacts()
 
 		// Get the number of languages for the contact
 		$slang = $db->getQuery(true)
-			->select('count(distinct(l.lang_code))')
+			->select('COUNT(DISTINCT l.lang_code)')
```
Comment (Quy) [inline administrator/components/com_languages/Helper/MultilangstatusHelper.php]: Apply quoteName

File: administrator/components/com_languages/Model/LanguagesModel.php [hunk 2]
```diff
@@ -112,46 +113,58 @@ protected function getListQuery()
...
+				]
+			)
+		)
+			->from($db->quoteName('#__languages', 'a'))
+			->join('LEFT', $db->quoteName('#__viewlevels', 'ag'), $db->quoteName('ag.id') . ' = ' . $db->quoteName('a.access'))
+			->join(
+				'LEFT',
+				$db->quoteName('#__menu', 'l'),
+				$db->quoteName('l.language') . ' = ' . $db->quoteName('a.lang_code')
+					. 'AND ' . $db->quoteName('l.home') . ' = 1 AND ' . $db->quoteName('l.language') . ' <> ' . $db->quote('*')
```
Comment (Quy) [inline administrator/components/com_languages/Model/LanguagesModel.php]:
```suggestion
					. ' AND ' . $db->quoteName('l.home') . ' = 1 AND ' . $db->quoteName('l.language') . ' <> ' . $db->quote('*')
```

File: administrator/components/com_languages/Helper/MultilangstatusHelper.php (hunk 1 above)
Comment (SharkyKZ) [reply administrator/components/com_languages/Helper/MultilangstatusHelper.php]: Applied.

File: administrator/components/com_languages/Model/LanguagesModel.php [hunk 3]
```diff
@@ -112,46 +113,58 @@ protected function getListQuery()
...
-		// Select the language home pages.
-		$query->select('l.home AS home')
-			->join('LEFT', $db->quoteName('#__menu') . ' AS l  ON  l.language = a.lang_code AND l.home=1  AND l.language <> ' . $db->quote('*'));
+		$query->select(
+			$this->getState('list.select',
+				[
+					$db->quoteName('a') . '.*',
+					$db->quoteName('l.home'),
+					$db->quoteName('ag.title', 'access_level'),
+
```
Comment (Quy) [inline administrator/components/com_languages/Model/LanguagesModel.php]:
```suggestion
```

**Comment by Quy** [comment]
I have tested this item :white_check_mark: successfully on f2ff8039818a8bb4452ba59b451f1778ecbd96d7<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/27248">issues.joomla.org/tracker/joomla-cms/27248</a>.</sub>

**Comment by jwaisner** [comment]
I have tested this item :white_check_mark: successfully on f2ff8039818a8bb4452ba59b451f1778ecbd96d7<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/27248">issues.joomla.org/tracker/joomla-cms/27248</a>.</sub>

**Comment by Quy** [comment]
RTC<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/27248">issues.joomla.org/tracker/joomla-cms/27248</a>.</sub>

**Comment by HLeithner** [comment]
Thanks

**Pull request merged**